### ISSUE TITLE
Questions about differences between the codes and the description in the paper

### ISSUE THREAD

**Issue by 50min** [LTTM/SDR#1]
Hello, thank you for sharing your codes. 

I have questions about DeepLab and learning rates.

According to the paper, SDR adopts DeepLab-V3+ and an initial learning rate is set to 1e-2 / 1e-3 for the first step / following steps, respectively.

However, in the provided code, SDR uses DeepLab-V3 as in 'MiB' and the initial learning rate is set to 1e-3 / 1e-4 for the first step / following steps, respectively.

Could you please clarify this? Thank you again

**Comment by SkyLLine** [comment]
thx a lot for your sharing, i've received the mail. Wish you good luck.

**Comment by umbertomichieli** [comment]
Hi, thank you very much!

The correct learning rates are:
- PASCAL VOC2012: 1e-3 / 1e-4
- ADE20K: 1e-2 / 1e-3

My apologies for not precisely stating it in the paper (and for the late reply). 
Hope this helps.

I'm closing the issue for now, feel free to re-open it.

**Comment by zhaoedf** [comment]
> Hi, thank you very much!
> 
> The correct learning rates are:
> 
> * PASCAL VOC2012: 1e-3 / 1e-4
> * ADE20K: 1e-2 / 1e-3
> 
> My apologies for not precisely stating it in the paper (and for the late reply). Hope this helps.
> 
> I'm closing the issue for now, feel free to re-open it.

but what about the V3 and V3+ issue? in paper, you said you use V3+, but in you code, you used V3.

**Comment by umbertomichieli** [comment]
Hi, I recall that we experimented with both DLV3 and DLV3+ and observed no clear winner among the two architectures in the various CL setups. This seems to be a typo in the paper; you can use DLv3 as reported in the codebase. Thank you for spotting this.

**Comment by zhaoedf** [comment]
> Hi, I recall that we experimented with both DLV3 and DLV3+ and observed no clear winner among the two architectures in the various CL setups. This seems to be a typo in the paper; you can use DLv3 as reported in the codebase. Thank you for spotting this.


sdr uses unkd in mib, right？ so no matter it is the v3 or v3 plus，you all follows mib and only distillates the head output? is this correct?

**Comment by umbertomichieli** [comment]
Yes, you are correct. In general, we found the output-level knowledge distillation (i.e., KD of the softmax probabilities) to be very simple and effective. We use unkd from MiB to naturally deal with the background shift issue when using disjoint/overlapped setups.

**Comment by jonasB00** [comment]
> Hi, I recall that we experimented with both DLV3 and DLV3+ and observed no clear winner among the two architectures in the various CL setups. This seems to be a typo in the paper; you can use DLv3 as reported in the codebase. Thank you for spotting this.

So just to be sure. The uploaded Code uses deeplab V3, but you have experimented on Deeplab V3 and Deeplab V3+.  The results that you gave in the paper are on Deeplav V3+, but for SDR there was no signifcant difference. 

Is that true? Im a bit confused now, because in the Paper you state, that you retrained MiB for fair comparison on Deeplab V3+.

**Comment by umbertomichieli** [comment]
Apologies for the confusion. The results reported in the paper are with DeepLab v3. You should not see significant differences when using DLv3+ in CL setting.

However, the backbone we used is the standard ResNet-101 ([from PyTorch](https://pytorch.org/vision/stable/models.html)). This is different from MiB, where they used [in place activated batch norm layers](https://github.com/mapillary/inplace_abn): the network structure is different and so are the starting pre-trained model weights. 

Therefore, we retrained MiB, as well as all the other approaches, on our same setup (network, hardware, etc).

**Comment by jonasB00** [comment]
okay, thank you so much. Just one last question :) Are the reported results trained on single GPU or on multiple GPUs?

**Comment by umbertomichieli** [comment]
The experiments have been carried out on a single GPU. This is another difference with respect to MiB, where they used distributed training.

**Comment by jonasB00** [comment]
okay, thank you for your quick answers!
